tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -526,36 +526,37 @@ struct mount {
+BPF_HASH(chosen_events_map, u32, u32);                          // events chosen by the user
+BPF_HASH(traced_pids_map, u32, u32);                            // track traced pids
+BPF_HASH(new_pids_map, u32, u32);                               // track processes of newly executed binaries
+BPF_HASH(containers_map, u32, u8);                              // map cgroup id to container status {EXISTED, CREATED, STARTED}
+BPF_HASH(args_map, u64, args_t);                                // persist args between function entry and return
+BPF_HASH(syscall_data_map, u32, syscall_data_t);                // persist data during syscall execution
+BPF_HASH(inequality_filter, u32, u64);                          // filter events by some uint field either by < or >
+BPF_HASH(uid_filter, u32, u32);                                 // filter events by UID, for specific UIDs either by == or !=
+BPF_HASH(pid_filter, u32, u32);                                 // filter events by PID
+BPF_HASH(mnt_ns_filter, u64, u32);                              // filter events by mount namespace id
+BPF_HASH(pid_ns_filter, u64, u32);                              // filter events by pid namespace id
+BPF_HASH(uts_ns_filter, string_filter_t, u32);                  // filter events by uts namespace name
+BPF_HASH(comm_filter, string_filter_t, u32);                    // filter events by command name
+BPF_HASH(bin_args_map, u64, bin_args_t);                        // persist args for send_bin funtion
+BPF_HASH(sys_32_to_64_map, u32, u32);                           // map 32bit to 64bit syscalls
+BPF_HASH(params_types_map, u32, u64);                           // encoded parameters types for event
+BPF_HASH(process_tree_map, u32, u32);                           // filter events by the ancestry of the traced process
+BPF_LRU_HASH(sock_ctx_map, u64, net_ctx_ext_t);                 // socket address to process context
+BPF_LRU_HASH(network_map, local_net_id_t, net_ctx_t);           // network identifier to process context
+BPF_ARRAY(file_filter, path_filter_t, 3);                       // filter vfs_write events
+BPF_ARRAY(string_store, path_filter_t, 1);                      // store strings from userspace
+BPF_PERCPU_ARRAY(bufs, buf_t, MAX_BUFFERS);                     // percpu global buffer variables
+BPF_PERCPU_ARRAY(bufs_off, u32, MAX_BUFFERS);                   // holds offsets to bufs respectively
+BPF_PROG_ARRAY(prog_array, MAX_TAIL_CALL);                      // store programs for tail calls
+BPF_PROG_ARRAY(prog_array_tp, MAX_TAIL_CALL);                   // store programs for tail calls
+BPF_PROG_ARRAY(sys_enter_tails, MAX_EVENT_ID);                  // store programs for tail calls
+BPF_PROG_ARRAY(sys_exit_tails, MAX_EVENT_ID);                   // store programs for tail calls
+BPF_STACK_TRACE(stack_addresses, MAX_STACK_ADDRESSES);          // store stack traces
+BPF_PERCPU_ARRAY(sock_common_struct, struct sock_common, 1);    // struct sock_common holder
```

Comment:
using a local variable reaches the BPF stack limit of 512 bytes on program tracepoint__inet_sock_set_state.
too many variables are saved on the stack. 
the way to avoid the limit is to use the ARRAY map, with the size of the struct, and read from there.

Reply:
If you will read the primitive fields and not the full struct, this should not be a problem